src/redux/users/tests/users-saga.test.js

```diff
@@ -0,0 +1,314 @@
+  handleUsersError
+} from '../users.saga';
+
+import { selectUsersAndTable } from '../../selectors/users.selectors';
+
+import {
+  mockUsersState,
+  mockTableState,
+  mockUsersList,
+  pageCount,
+  statuses,
+  adminInput,
+  mockUser,
+  mockAdmin,
+  mockToken,
+  mockError,
+  mockSnackarState,
+  mockStatus
+} from './users.variables';
+
+import {
+  setUsersLoading,
+  setUsers,
+  setUser,
+  setUserError,
+  deleteUserLocally,
+  updateUserLocally
+} from '../users.actions';
+
+import { setPagesCount, setItemsCount } from '../../table/table.actions';
+
+import {
+  setSnackBarMessage,
+  setSnackBarSeverity,
+  setSnackBarStatus
+} from '../../snackbar/snackbar.actions';
+
+import Users from '../users.reducer';
+import Table from '../../table/table.reducer';
+import Snackbar from '../../snackbar/snackbar.reducer';
+
+const {
+  SUCCESS_DELETE_STATUS,
+  SUCCESS_UPDATE_STATUS,
+  SUCCESS_CONFIRMATION_STATUS,
+  SUCCESS_CREATION_STATUS
+} = statuses;
+
+describe('Users saga test', () => {
+  it('should load all users', () =>
+    expectSaga(handleUsersLoad)
+      .withReducer(
+        combineReducers({
+          Users,
+          Table
+        }),
+        {
+          Users: mockUsersState,
+          Table: mockTableState
+        }
+      )
+      .provide([
+        [
+          select(selectUsersAndTable),
+          {
+            usersState: mockUsersState,
+            tableState: mockTableState
+          }
+        ],
+        [call(getAllUsers, mockUsersState, mockTableState), mockUsersList]
+      ])
+      .put(setUsersLoading(true))
+      .put(setPagesCount(pageCount))
+      .put(setItemsCount(mockUsersList.count))
+      .put(setUsers(mockUsersList.items))
+      .put(setUsersLoading(false))
+      .hasFinalState({
+        Users: {
+          ...mockUsersState,
+          list: mockUsersList.items
+        },
+        Table: {
+          ...mockTableState,
+          itemsCount: mockUsersList.count,
+          pagination: {
+            ...mockTableState.pagination,
+            pagesCount: pageCount
+          }
+        }
+      })
+      .run()
+      .then((result) => {
+        const { allEffects: analysis } = result;
+        const analysisPut = analysis.filter((e) => e.type === 'PUT');
+        expect(analysisPut).toHaveLength(5);
+      }));
+
+  it('should load all user by id', () =>
+    expectSaga(handleUserLoad, { payload: mockUser._id })
+      .withReducer(combineReducers({ Users }), { Users: mockUsersState })
+      .put(setUsersLoading(true))
+      .provide([[call(getUserById, mockUser._id), mockUser]])
+      .put(setUser(mockUser))
+      .put(setUsersLoading(false))
+      .hasFinalState({
+        Users: {
+          ...mockUsersState,
+          user: mockUser
+        }
+      })
+      .run()
+      .then((result) => {
+        const { allEffects: analysis } = result;
+        const analysisPut = analysis.filter((e) => e.type === 'PUT');
+        expect(analysisPut).toHaveLength(3);
```

Comment:
Why you did not check the length of `CALL`?

Reply:
good tip
I will add it